python/ray/experimental/serve/queues.py

```diff
@@ -220,3 +228,162 @@ def flush(self):
             self.self_handle._flush.remote()
         else:
             self._flush()
+
+
+class RandomPolicyQueue(CentralizedQueues):
+    """
+    A wrapper class for Random policy.
+    This backend selection policy is `Stateless` meaning
+    the current decisions of selecting backend are not
+    dependent on previous decisions.
+    """
+
+    def _flush_service_queue(self):
+        # perform traffic splitting for requests
+        for service, queue in self.queues.items():
+            # while there are incoming requests and there are backends
+            while len(queue) and len(self.traffic[service]):
+                backend_names = list(self.traffic[service].keys())
+                backend_weights = list(self.traffic[service].values())
+                # randomly choose a backend for every query
+                chosen_backend = np.random.choice(
+                    backend_names, p=backend_weights).squeeze()
+
+                request = queue.popleft()
+                self.buffer_queues[chosen_backend].add(request)
+
+
+@ray.remote
+class RandomPolicyQueueActor(RandomPolicyQueue, CentralizedQueuesActor):
+    pass
+
+
+class RoundRobinPolicyQueue(CentralizedQueues):
+    """
```

Comment:
So traffic policy weights has no use here. We might explicitly say that or use some sort of weighted round robin?

Reply:
Yeah this queueing policy does not use traffic weights. Weighted roundRobin can be implemented as a separate policy